src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -916,9 +916,49 @@ void Battleground::EndBattleground(PvPTeamId winnerTeamId)
         player->UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_COMPLETE_BATTLEGROUND, player->GetMapId());
     }
 
+    if (IsEventActive(EVENT_SPIRIT_OF_COMPETITION) && isBattleground())
+        SpiritofCompetitionEvent(winnerTeamId);
+
     sScriptMgr->OnBattlegroundEnd(this, GetTeamId(winnerTeamId));
 }
 
+bool Battleground::SpiritofCompetitionEvent(PvPTeamId winnerTeamId)
+{
+    // Everyone is eligible for tabard reward
+    for (BattlegroundPlayerMap::const_iterator itr = m_Players.begin(); itr != m_Players.end(); ++itr)
+    {
+        Player* player = itr->second;
+        bool questStatus = player->GetQuestStatus(QUEST_FLAG_PARTICIPANT) != QUEST_STATUS_REWARDED;
+        if (player && questStatus)
+            player->CastSpell(player, SPELL_SPIRIT_OF_COMPETITION_PARTICIPANT, true);
+    }
+
+    // in case of draw nobody get reward
```

Comment:
```suggestion
    // In case of a draw nobody get rewarded
```

Reply:
maybe bench will complain about this too kek

Reply:
`In case of a draw, nobody gets rewarded`